File: lib/items.dart
import 'package:scrapy/scrapy.dart';

class IpProxy extends Item {
  String ip;
  String port;
  String location;
  String protocol;

  IpProxy({this.ip, this.port, this.location, this.protocol});

  @override
  String toString() {
    return "Proxy : { ip : $ip, port: $port, location: $location, protocol: $protocol }";
  }

  Map<String, dynamic> toJson() =>
      {'ip': ip, 'port': port, 'location': location, 'protocol': protocol};
}

class IpProxies<IpProxy> extends Items {
  @override
  Map<String, dynamic> toJson() {
    return super.toJson();
  }
}

File: scrapy-python/proxylist/items.py
# -*- coding: utf-8 -*-

# Define here the models for your scraped items
#
# See documentation in:
# https://doc.scrapy.org/en/latest/topics/items.html

import scrapy


class IPlistItem(scrapy.Item):
    # define the fields for your item here like:
    ip = scrapy.Field()
    port = scrapy.Field()
    protocol = scrapy.Field()
    location = scrapy.Field()
